docs/disconnected-updateservice-operator.md

```diff
@@ -0,0 +1,295 @@
+
+In order to build a container image that contains the graph-data, which will be used as init
+container in the updateservice pod: 
+
+```
+wget https://github.com/openshift/cincinnati-graph-data/archive/master.tar.gz -O graph.tar.gz
+cat <<EOF > Dockerfile
+FROM registry.access.redhat.com/ubi8/ubi:8.1
+COPY graph.tar.gz /tmp/
+RUN mkdir -p /var/lib/cincinnati/graph-data/
+CMD exec /bin/bash -c "tar xvzf /tmp/graph.tar.gz -C /var/lib/cincinnati/graph-data/ --strip-components=1"
+EOF
+
+podman build -f Dockerfile -t ${DISCONNECTED_REGISTRY}/updateservice/graph-data:v1
+podman push ${DISCONNECTED_REGISTRY}/updateservice/graph-data:v1
+```
+## Mirror the release image and the release content
+
+You might want to review the documentation around disconnected registries to learn about mirroring OpenShift Releases.
+
+```
+export UPSTREAM_REGISTRY='quay.io'
+export PRODUCT_REPO='openshift-release-dev'
+export RELEASE_NAME='ocp-release'
+export OCP_RELEASE='4.5.2-x86_64'
+export LOCAL_REGISTRY='my-disconnected-registry.example.com:5000'
+export LOCAL_SECRET_JSON='/path/to/pull_secret.json'
+
+oc adm -a ${LOCAL_SECRET_JSON} release mirror \
+--from=${UPSTREAM_REGISTRY}/${PRODUCT_REPO}/${RELEASE_NAME}:${OCP_RELEASE} \
+--to=${LOCAL_REGISTRY}/ocp4 \
+--to-release-image=${LOCAL_REGISTRY}/ocp4/release:${OCP_RELEASE}
+
+export OCP_RELEASE='4.5.3-x86_64'
+
+oc adm -a ${LOCAL_SECRET_JSON} release mirror \
+--from=${UPSTREAM_REGISTRY}/${PRODUCT_REPO}/${RELEASE_NAME}:${OCP_RELEASE} \
+--to=${LOCAL_REGISTRY}/ocp4 \
+--to-release-image=${LOCAL_REGISTRY}/ocp4/release:${OCP_RELEASE}
+```
+
+
+## Deploy UpdateService instance
+
+```
+cat create-updateservice-instance.yaml
+---
+apiVersion: updateservice.operator.openshift.io/v1
+kind: UpdateService
+metadata:
+  name: karampok
+  namespace: openshift-update-service
+spec:
+  replicas: 1
+  releases: "{DISCONNECTED_REGISTRY}/ocp4/release"
+  #releases: "{DISCONNECTED_REGISTRY}/ocp4 // OCP Release Image and Release Content in the same path
+  graphDataImage: "{DISCONNECTED_REGISTRY}/updateservice/graph-data:v4"
+```
+    
+### Known Issues
+
+#### OOMKilled
+
+If you have you mirrored your OpenShift Release Image and Release Content
+with an identical path, UpdateService's Graph Builder will take more memory
+than allowed by the Deployment configuration and the pod will be OOMKilled.
+If this is your case you can copy the release image to a different namespace
+in your registry.
+
+```
+skopeo copy docker://${DISCONNECTED_REGISTRY}/ocp4:4.5.2-x86_64 docker://${DISCONNECTED_REGISTRY}/release:4.5.2-x86_64 --authfile=/path/to/pull_secret.json
+skopeo copy docker://${DISCONNECTED_REGISTRY}/ocp5:4.5.3-x86_64 docker://${DISCONNECTED_REGISTRY}/release:4.5.3-x86_64 --authfile=/path/to/pull_secret.json
+```
+
+#### CreateRouteFailed
+
+    
+If you see `ReconcileCompleted` status as `false` with reason `CreateRouteFailed` caused by `host must conform to DNS 1123 naming convention`
```

Comment:
Is it worth mentioning? It looks like a fairly common error

Reply:
This depends on the base domain of the cluster, it's a limitation on the route names length. I believe keeping it here doesn't hurt, but up to you.

Reply:
Sounds like its a common problem, so lets keep it - it needs a link to bugzilla though

Reply:
If operator allows them to configure settings which end up in an invalid route I'd say its a bug

Reply:
I do not think there is bug here, if I understand it is just a small reminder that 
basedomain + object name is the DNS name which should feed the length.